apps/frontend/src/pages/tasks/libs/components/deadline/deadline.tsx

```diff
@@ -26,57 +27,61 @@ const Deadline: React.FC<Properties> = ({ deadline }: Properties) => {
 		isExpired && styles["expired"],
 	);
 
-	const calculateDaysUntilDeadline = useCallback((): boolean => {
-		const deadlineTime = new Date(deadline).getTime();
-		const currentTime = Date.now();
-		const timeToDeadline = deadlineTime - currentTime;
-
-		if (timeToDeadline < DEADLINE_OVER) {
-			setCountdown(COUNTDOWN_EXPIRED);
-			setIsExpired(true);
-
-			return true;
-		}
-
-		const days = Math.floor(timeToDeadline / MillisecondsPerUnit.DAY);
-		const hours = Math.floor(
-			(timeToDeadline % MillisecondsPerUnit.DAY) / MillisecondsPerUnit.HOUR,
-		);
-		const minutes = Math.floor(
-			(timeToDeadline % MillisecondsPerUnit.HOUR) / MillisecondsPerUnit.MINUTE,
-		);
-
-		const formattedDays = String(days);
-		const formattedHours = String(hours).padStart(TimePad.HOURS, TIME_PAD_FILL);
-		const formattedMinutes = String(minutes).padStart(
-			TimePad.MINUTES,
-			TIME_PAD_FILL,
-		);
-
-		setCountdown({
-			days: formattedDays,
-			hours: formattedHours,
-			minutes: formattedMinutes,
-		});
-
-		return false;
-	}, [deadline]);
-
 	useEffect(() => {
+		const calculateDaysUntilDeadline = (): boolean => {
```

Comment:
Why not move to a separate helper? I don't like the fact, that it is defined inside of useEffect, which means it will be created each time useEffect runs